compiler/src/org.graalvm.compiler.serviceprovider/src/org/graalvm/compiler/serviceprovider/GraalServices.java

```diff
@@ -244,4 +244,13 @@ public static double fma(double a, double b, double c) {
     public static VirtualObject createVirtualObject(ResolvedJavaType type, int id, boolean isAutoBox) {
         throw shouldNotReachHere();
     }
+
+    /**
+     * Gets the update-release counter for the current Java runtime.
+     *
+     * @see "https://download.java.net/java/GA/jdk14/docs/api/java.base/java/lang/Runtime.Version.html"
+     */
+    public static int getJavaUpdateVersion() {
+        throw shouldNotReachHere();
```

Comment:
Why should not reach here? What Java versions does this one deal with? 14+?

Reply:
@galderz This is a backport from master. See https://github.com/oracle/graal/pull/2403 and https://github.com/oracle/graal/issues/2402

To answer your question, this is a `shouldNotReachHere()` as there must be JDK version specific overrides which should be called instead.